vivaldi/hooks/treeview-spacing.js

```diff
@@ -0,0 +1,87 @@
+// Change spacing of tree views (may need additional CSS)
+
+vivaldi.jdhooks.addStyle(`
+    :root {
+        --rowHeight: 24px;
+    }
+    .vivaldi-tree .tree-item .tree-row label {
+        line-height: var(--rowHeight);
+        min-height: var(--rowHeight);
+    }
+    .vivaldi-settings .settings-sidebar .button-category {
+        height: var(--rowHeight);
+    }
+`, 'treeview-spacing.js');
+
+vivaldi.jdhooks.hookModule("vivaldiSettings", (moduleInfo, exports) => {
+    let oldGetDefault = exports.getDefault
+    exports.getDefault = name => {
+        switch (name) {
+            case "VIVALDI_TREE_ROW_HEIGHT": return 24
+            default: return oldGetDefault(name)
+        }
+    }
+    return exports
+})
+
+vivaldi.jdhooks.hookClass('common_VivaldiTreeList', cls => {
+    const newCls = vivaldi.jdhooks.insertWatcher(class extends cls {
+        render() {
+            this.props.rowHeight = this.state.jdVivaldiSettings.VIVALDI_TREE_ROW_HEIGHT
```

Comment:
So what you want me to do is to hook all the classes that use it (should be 7 in total), right? I think adding settings per tree type would be easier and better for users (coz one may want different size in bookmarks & in notes etc.).

I’m not sure if (& how) your suggestions would work. How do you mean the function? That would have to be called by me, or not?

Reply:
I proposed 2 ways:
- change only that items which height is equal to default
- use multiplication by some floating value to change height of every item

May be there are other solutions exist.

Talking about a function I mean `props.rowHeight` may be a function returning height by item's index, so items of the same tree may have different heights. So you need to ignore values if their `typeof` is "function" or make overridden value a function calling original one and modifying its return value. 
I haven't seen this feature is used anywhere but may be it will appear when mail or some other new feature is published.

Reply:
Sorry for the delay, I’m quite unfocused recently. It seems that I can only change the spacing from the `TreeList` itself, so I have to find a way to differentiate the trees—it seems *most* of them have either unique `props.mimeType` or `props.settings`, so I think this is the way to go. If you have any suggestions, please let me know.

(Just a progress report)

Reply:
Can it be like this? https://github.com/justdanpo/VivaldiHooks/pull/37/commits/40778aefd6de6f6f8555ae07a4f82ebe6cc5871a